packtools/sps/formats/oai_dc.py

```diff
@@ -0,0 +1,561 @@
+
+def add_datestamp(header):
+    el = ET.Element('datestamp')
+    el.text = get_datestamp()
+    header.append(el)
+
+
+def add_set_spec(header, xml_tree):
+    issn = get_issn(xml_tree)
+    if issn != '':
+        el = ET.Element('setSpec')
+        el.text = issn
+        header.append(el)
+
+
+def get_issn(xml_tree):
+    issns = journal_meta.ISSN(xml_tree)
+
+    return issns.epub or issns.ppub
+
+
+def add_title(xml_oai_dc, title):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}title')
+    el.text = title.strip()
+
+    xml_oai_dc.append(el)
+
+
+def add_creator(xml_oai_dc, author_name):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}creator')
+    el.text = author_name.strip()
+
+    xml_oai_dc.append(el)
+
+
+def add_subject(xml_oai_dc, kw, article):
+    if kw.get('lang') == article.main_lang:
+        el = ET.Element('{http://purl.org/dc/elements/1.1/}subject')
+        el.text = kw.get('text').strip()
+
+        xml_oai_dc.append(el)
+
+
+def get_description(abstract):
+    try:
+        description = [abstract.main_abstract_with_tags['title']]
+        for key, value in abstract.main_abstract_with_tags['sections'].items():
+            description.append(key)
+            description.append(value)
+
+        return " ".join(description)
+    except IndexError:
+        pass
+
+
+def add_description(xml_oai_dc, description):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}description')
+    el.text = description.strip()
+
+    xml_oai_dc.append(el)
+
+
+def add_publisher(xml_oai_dc, publisher):
+    try:
+        el = ET.Element('{http://purl.org/dc/elements/1.1/}publisher')
+        el.text = publisher.publishers_names[0].strip()
+
+        xml_oai_dc.append(el)
+    except IndexError:
+        pass
+
+
+def add_source(xml_oai_dc, journal):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}source')
+    el.text = journal.journal_title.strip()
+
+    xml_oai_dc.append(el)
+
+
+def get_date(dt):
+    year = dt.pub_dates[0].get('year')
+    month = dt.pub_dates[0].get('month')
+    day = dt.pub_dates[0].get('day')
+    exceptions = [None, '', '0', '00']
+
+    if year is None:
+        return
+    if month is None and day is None:
+        return year
+    month = '01' if month in exceptions else month
+    day = '01' if day in exceptions else day
+
+    return '-'.join([year, month, day])
+
+
+def add_date(xml_oai_dc, dt_out):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}date')
+    el.text = dt_out
+    xml_oai_dc.append(el)
+
+
+def add_uri_identifier(xml_oai_dc, identifier):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}identifier')
+    el.text = identifier.get('sci_arttext')
+    xml_oai_dc.append(el)
+
+
+def add_lang(xml_oai_dc, lang):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}language')
+    el.text = lang.main_lang
+    xml_oai_dc.append(el)
+
+
+def add_relation(xml_oai_dc, related_article):
+    el = ET.Element('{http://purl.org/dc/elements/1.1/}relation')
```

Comment:
Esta linha tem que ficar dentro do `for`

Reply:
@Rossi-Luciano mova o conteúdo desta função no local onde ocorre a chamada e elimine esta função

Reply:
Ok @robertatakenaka